config/package_resource.go

```diff
@@ -16,18 +16,28 @@ package config
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
 	"time"
 
+	"github.com/GoogleCloudPlatform/osconfig/agentconfig"
 	"github.com/GoogleCloudPlatform/osconfig/clog"
 	"github.com/GoogleCloudPlatform/osconfig/packages"
 	"github.com/GoogleCloudPlatform/osconfig/util"
 
 	agentendpointpb "google.golang.org/genproto/googleapis/cloud/osconfig/agentendpoint/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	packageInfoCacheFile = filepath.Join(agentconfig.CacheDir(), "config_package_info.cache")
+	// Clear out the entry if the last lookup is > 7 days ago.
```

Comment:
Should we be using one of the standard supported cache packages listed below instead of implementing one.

The standard packages should help to avoid running into memory issues if there are too many entries in the map within this time period

https://g3doc.corp.google.com/util/cache/go/README.md?cl=head

WDYT?

Reply:
Those are in memory caches (and not opened source), we don't need a complex cache, just a local set of files. File size should really not be an issue because the agent will already be hold this in memory (it's just the API calls). I have an idea on how we can reduce file sizes though (one cache per assignment) that I am going to look at.